R/rcpp.R

```diff
@@ -109,3 +107,18 @@ use_makevars <- function(settings = NULL) {
     edit_file(makevars_win_path)
   }
 }
+
+use_src_example_script <- function(name, src_type = c("cpp", "c") ) {
+
+  src_type = match.arg(src_type)
+
+  if (is.null(name) && !check_directory_has_files(path("src"))) {
```

Comment:
I'm surprised this works because our checkers, in general, either don't return anything or return their input. In general, they don't return logical.

I'm happy to take a turn with this (fiddle with directory utilities, that aren't just for this) and then push a few commits in and see if you like it. Sound ok?

Reply:
Sounds perfect. Though, I did add `check_direct_has_files()`, c.f. https://github.com/r-lib/usethis/pull/860/commits/6957e4836dc74f0e6ef6696a6d4ac61438817714

Reply:
OK I'll make some changes in this PR and tag you to take a look.